docs/source/configuration.rst

```diff
@@ -14,11 +15,135 @@ After you have installed `Weaver`, you can customize its behaviour using multipl
+
+Most configuration parameters for the *manager* portion of `Weaver` (i.e.: WSGI HTTP server for API endpoints) are
+defined in the ``[app:main]`` section of `weaver.ini.example`_, while parameters specific to the *worker* (task queue
+handler) are within ``[celery]`` section. Note that multiple settings are shared between the two applications, such as
+the ``mongodb.[...]`` configuration or ``weaver.configuration`` options. When parameters are shared, they are usually
+expected to be placed in ``[app:main]`` section.
+
+Following is a partial list of most predominant settings specific to `Weaver`.
+
+- | ``weaver.configuration = ADES|EMS``
+  |
+  | Tells the application in which mode to run.
+  |
+  | Enabling `ADES` for instance will disable some `EMS`-specific
+    operations such as dispatching :ref:`Workflow` process steps to known remote `ADES` servers.
+
+- | ``weaver.url = <url>``
+  |
+  | Defines the full URL (including HTTP protocol/scheme, hostname and optionally additional path suffix) that will
+    be used as base URL for all other URL settings of `Weaver`.
+
+.. note::
+
+    This is the URL that you want displayed in responses (e.g.: ``processDescriptionURL`` or job ``location``).
+    For the effective URL employed by the WSGI HTTP server, refer to ``[server:main]`` section of `weaver.ini.example`_.
+
+- | ``weaver.wps = true|false``
+  |
+  | Enables the WPS-1/2 endpoint.
+
+ .. warning::
+
+     At the moment, this setting must be ``true`` to allow job execution as the worker monitors this endpoint.
+     This could change with future developments (see issue `#21 <https://github.com/crim-ca/weaver/issues/21>`_).
+
+- | ``weaver.wps_path = <url-path>``
+  | ``weaver.wps_url = <full-url>``
+  | (default: *path* ``/ows/wps``)
+  |
+  | Defines the URL to employ as WPS-1/2 endpoint.
+  |
+  | It can either be the explicit *full URL* to use or the *path* relative to ``weaver.url``.
+  | Setting ``weaver.wps_path`` is ignored if its URL equivalent is defined.
+  | The *path* variant **SHOULD** start with ``/`` for appropriate concatenation with ``weaver.url``, although this is
+    not strictly enforced.
+
+- | ``weaver.wps_output_dir = <directory-path>``
+  | (default: ``/tmp``)
+  |
+  | Location where WPS outputs (results from jobs) will be stored for stage-out.
+  |
+  | This directory should be mapped to `Weaver`'s WPS output URL to serve them externally as needed.
+
+- | ``weaver.wps_output_path = <url-path>``
+  | ``weaver.wps_output_url = <full-url>``
+  | (default: *path* ``/wpsoutputs``)
+  |
+  | Endpoint that will be employed as prefix to refer to WPS outputs (job results).
+  |
+  | It can either be the explicit *full URL* to use or the *path* relative to ``weaver.url``.
+  | Setting ``weaver.wps_output_path`` is ignored if its URL equivalent is defined.
+  | The *path* variant **SHOULD** start with ``/`` for appropriate concatenation with ``weaver.url``, although this is
+    not strictly enforced.
+
```

Comment:
Would it be better to enforce it, instead of having runtime issues if this detail is missing? Either by giving a warning when config is getting read or by adding the '/' if absent.

Reply:
Warning could be a good idea, but since this setting is read really often, I don't want to spam too much logs either. We could need to have smart warning that disables itself after the first occurrence. 

We shouldn't enforce since the path without `/` is valid, it really depends on how `weaver.url` is defined. For example, if someone prefers to use `weaver.url` finishing with `/`, enforcing one in `weaver.wps_output_path` would add it twice.

Reply:
I see. Therefore adding or removing the / could be during the merge or url/path. Not a priority for now but a simple detail that can create error/confusion as the platform grows and more technicalities about urls are defined (eg: this one is / prefixed, the other one / suffixed, this url path no slash, this one prefixed and suffixed).. If we have 3-4 urls/paths well I think that's ok but as the config needs grows, it could be helpful to handle those cases.

Reply:
I prefer to avoid as much as possible to add/remove anything inplace. If the config is invalid, the user should simply update it to make is correct. When the config is invalid, usually you can see it rather quickly as something is not working as expected.

I have had too many situations when an extra or missing `/` caused me so many problems within nginx configurations. I'd personally rather know that when I explicitly set a `/` at some location (regardless where it is), it is applied as is and the code doesn't "help me" undo my configuration.

Reply:
Good point, then specifying the url format in the doc as you did is perfect.